test-data/unit/check-inference.test

```diff
@@ -686,7 +686,7 @@ g('a')() # E: List[str] not callable
 # call, would be preferable here.
 g(['a']) # E: Argument 1 to "g" has incompatible type List[str]; expected List[<uninhabited>]
 
-h(g(['a']))
+h(g(['a'])) # E: Argument 1 to "g" has incompatible type List[str]; expected List[<uninhabited>]
```

Comment:
This looks like a regression, and I suspect that there are other cases that could be affected. By special casing the fix to lambda expressions this should hopefully be fixed.